docs/docs/service-discovery-load-balancing.md

```diff
@@ -6,15 +6,37 @@ title: Service Discovery & Load Balancing
 
 Once our app is up and running, we need a way to send outside traffic to it. If we're running multiple apps, they also need a way to find each other. We can use [Mesos-DNS](https://github.com/mesosphere/mesos-dns) for service discovery through the domain name system ([DNS](http://en.wikipedia.org/wiki/Domain_Name_System)). Mesos-DNS generates a hostname for each application running on Mesos and translates these names to the IP address and port on the machine currently running each application. Mesos-DNS is particularly useful if the apps are launched through multiple frameworks (not just Marathon). Check the Mesos-DNS [documentation and tutorials page](http://mesosphere.github.io/mesos-dns/) for further information. 
 
-An alternative way for service discovery is to run a TCP proxy on each host in the cluster, and transparently forward a static port on localhost to the hosts that are running the app. That way, clients simply connect to that port, and the implementation details of discovery are completely abstracted away. This approach is sufficient if all apps are launched through Marathon. 
-
-## Ports Assignment
-
-When an application is created in Marathon (either through the REST API or the front end) - you may assign it one or more ports. These ports may be any valid port number or 0, which Marathon will use to randomly assign a port number.
-
-This port is used to ensure no two applications can be run using Marathon with overlapping port assignments (i.e. that the ports are globally unique).
-
-However, this is not the port that each instance of that application is assigned when running on slave nodes within the Mesos cluster. Since multiple instances may run on the same node, each instance (or *task*) is assigned a random port. This can be accessed from the `$PORT` environment variable which is set by Marathon.
+An alternative way to implement service discovery is to run a TCP proxy on each host in the cluster, and transparently forward 
+connections to the static service port on localhost to the dynamically assigned host/port combinations of the individual 
+app instances (= *tasks*). That way, clients simply connect to that port, and the implementation details of discovery 
+are completely abstracted away. This approach is sufficient if all apps are launched through Marathon. 
+
+## Service Ports Assignment
+
+When Marathon creates an application in Marathon (either through the REST API or the front end),
+you may assign one or more service ports to it. 
+You can specify an valid port numbers as service ports or you can use 0 to indicate that Marathon should assign
```

Comment:
s/an valid port numbers/valid port numbers/ ?

Maybe we should specify that a "valid port number" is a number greater than 0 and that they must be unique across all apps.